oasis-reference/src/test/kotlin/com/empowerops/volition/ref_oasis/SystemTest.kt

```diff
@@ -0,0 +1,666 @@
+            assertThat(it.message).isEqualTo(buildAutoSetupMessage(testSimulation, true))
+        })
+        assertThat(logger.log.toDebugString()).isEqualTo("""
+            |Optimizer Event > ${PluginRegisteredEvent(name = "N1")}
+            |Optimizer Event > ${PluginUpdatedEvent(name = "N1")}
+            |Optimizer Event > ${ProxyAddedEvent(name = "N1")}
+            |Optimizer Event > ${ProxyUpdatedEvent(name = "N1")}
+        """.trimMargin())
+    }
+
+    @Test
+    fun `auto setup but not register should deny`() {
+        //deny
+    }
+
+    @Test
+    fun `when send update with no proxy should do the update only the sim`() {
+        //setup
+        val endpoint = create()
+        endpoint.registerRequest(registerDTO, mock())
+        val responseObserver = mock<StreamObserver<NodeChangeConfirmDTO>>()
+
+        //act
+        endpoint.updateNode(testStatusRequest, responseObserver)
+
+        //assert
+        verify(responseObserver, times(1)).onNext(check {
+            assertThat(it.message).isEqualTo(buildSimulationUpdateMessage(testSimulation, true))
+        })
+        assertThat(logger.log.toDebugString()).isEqualTo("""
+            |Optimizer Event > ${PluginRegisteredEvent(name = "N1")}
+            |Optimizer Event > ${PluginUpdatedEvent(name = "N1")}
+        """.trimMargin())
+    }
+
+    @Test
+    fun `when send update with proxy should do the update both`() {
+        //setup
+        val endpoint = create()
+        endpoint.registerRequest(registerDTO, mock())
+        endpoint.autoConfigure(testStatusRequest, mock())
+        val responseObserver = mock<StreamObserver<NodeChangeConfirmDTO>>()
+
+        //act
+        endpoint.updateNode(testStatusRequest2, responseObserver)
+
+        //assert
+        verify(responseObserver, times(1)).onNext(check {
+            assertThat(it.message).isEqualTo(buildSimulationUpdateMessage(testSimulation2, true))
+        })
+        assertThat(logger.log.toDebugString()).isEqualTo("""
+            |Optimizer Event > ${PluginRegisteredEvent(name = "N1")}
+            |Optimizer Event > ${PluginUpdatedEvent(name = "N1")}
+            |Optimizer Event > ${ProxyAddedEvent(name = "N1")}
+            |Optimizer Event > ${ProxyUpdatedEvent(name = "N1")}
+            |Optimizer Event > ${PluginUpdatedEvent(name = "N1")}
+            |Optimizer Event > ${ProxyUpdatedEvent(name = "N1")}
+        """.trimMargin())
+    }
+
+
+    val startRequest = StartOptimizationCommandDTO.newBuilder().setName("N3").build()
+    val resultResponse = SimulationResponseDTO.newBuilder().setName("N3").build()
+    val errorResponse = ErrorResponseDTO.newBuilder().setName("N3").setMessage("Test error").setException("Test exception").build()
+    val stopRequest = StopOptimizationCommandDTO.newBuilder().setName("N3").build()
+    val expectedInputs = mapOf("x1" to 5.0, "x2" to 5.0, "x3" to 5.0)
+
+    /**
+     * Execution: Start/ Stop
+     */
+    @Test @Disabled
```

Comment:
This is not the only one i am having problem with right now. The channel implementation looks logical to me but I am not sure where all the hanging coming from since I wasn't able to replicate them consistently. 

Another thing is that it is hard to assert on the log for some action due to the same reason. Things can happen at any differnt time but our state machine is doing a good job not letting thing happend when it is not suppose to. This make it hard until we have a state machine that is awatiable so we can make sure we are doing the correct action at the correct time.

Reply:
Also, the system test so far is lacking on the assertion of the end state